src/TaskRunner/Commands/ToolCommands.php

```diff
@@ -405,23 +405,26 @@ protected function componentInsecure($modules)
         } else {
             $insecurePackages = json_decode($result, true);
             if (is_array($insecurePackages)) {
+                $messages = [];
                 foreach ($insecurePackages as $insecurePackage) {
-                    $historyTerms = $this->getPackageDetails($insecurePackage['name'], $insecurePackage['version'], '8.x');
-                    $packageInsecureConfirmation = true;
                     $msg = "Package {$insecurePackage['name']} have a security update, please update to a safe version.";
-
-                    if (empty($historyTerms['terms']) || !in_array("insecure", $historyTerms['terms'])) {
-                        $packageInsecureConfirmation = false;
-                        $msg .= ' (Confirmation failed, ignored)';
-                    }
                     if (!empty($modules[$insecurePackage['name']]['secure'])) {
                         if (Semver::satisfies($insecurePackage['version'], $modules[$insecurePackage['name']]['secure'])) {
-                            $packageInsecureConfirmation = false;
-                            $msg .= ' (Version marked as secure)';
+                            $messages[] = "$msg (Version marked as secure)";
+                            continue;
                         }
                     }
-                    echo $msg . PHP_EOL;
-                    $this->componentCheckInsecureFailed = $packageInsecureConfirmation;
+                    $historyTerms = $this->getPackageDetails($insecurePackage['name'], $insecurePackage['version'], '8.x');
+                    if (empty($historyTerms['terms']) || !in_array('insecure', $historyTerms['terms'])) {
+                        $messages[] = "$msg (Confirmation failed, ignored)";
+                        continue;
+                    }
+
+                    $messages[] = $msg;
+                    $this->componentCheckInsecureFailed = true;
```

Comment:
Not clear to me where you will in fact block it. As much I can see failing the confirmation or the existence of a secure version will not change the status.

Reply:
Not sure if I understood the question.
Every package that enters in the foreach is already insecure, then we have exceptions, if not exception is found the package is marked as insecure.

Reply:
ok I think I understood your question, the return "fail" is thrown in the main function 
https://github.com/ec-europa/toolkit/pull/507/files#diff-e8dec610a839211696c5aa5165037fc2369ee50f283b9527877261c873c72d23R201